sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -2561,6 +2563,40 @@ class DatasetSuite extends QueryTest
 
     checkDataset(ds.filter(f(col("_1"))), Tuple1(ValueClass(2)))
   }
+
+  test("CLASS_UNSUPPORTED_BY_MAP_OBJECTS when creating dataset") {
```

Comment:
Not sure is this the best place for this test case, currently we don't have test case for creating dataset using `spark.createDatSet`, so I add it to here.